### PR TITLE
DM/Openapi: use reverse proxy instead of redirect

### LINKED ISSUE #5179: dm openapi should use reverse proxy instead of redirect
### Is your feature request related to a problem?

user deploy three dm-master  in k8s and use k8s-svc to expose the dm-master addr

if user's request was routed to a non-leader node, dm-master will redirct this request to leader node, 
but when the user is isolated from the current leader node, the request cannot be processed properly

### Describe the feature you'd like

use reverse proxy instead of redirct when a request is send to non-leader node

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

> By default, kube-proxy in userspace mode chooses a backend via a round-robin algorithm

this means user request will be split evenly between the there dm-master pod , see more in https://kubernetes.io/docs/concepts/services-networking/service/

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Issue Number: close #5179

### What is changed and how it works?
use reverse proxy instead of redirect.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by D3Hunter — APPROVED** [review]

File: dm/dm/master/openapi_view.go [hunk 1]
```diff
@@ -60,6 +61,34 @@ func (s *Server) redirectRequestToLeaderMW() gin.HandlerFunc {
 	}
 }
 
+// reverseRequestToLeaderMW reverses request to leader.
+func (s *Server) reverseRequestToLeaderMW() gin.HandlerFunc {
```
Comment (Ehco1996) [inline dm/dm/master/openapi_view.go:65]: please delete the deadcode `redirectRequestToLeaderMW `

File: dm/dm/master/openapi_view.go [hunk 2]
```diff
@@ -60,6 +61,34 @@ func (s *Server) redirectRequestToLeaderMW() gin.HandlerFunc {
...
+			// nolint:dogsled
+			_, _, leaderOpenAPIAddr, err := s.election.LeaderInfo(ctx2)
+			if err != nil {
+				_ = c.AbortWithError(http.StatusBadRequest, err)
+				return
+			}
+			// simpleProxy just reverses to leader host
+			simpleProxy := httputil.ReverseProxy{
+				Director: func(req *http.Request) {
+					req.URL.Scheme = "http"
```
Comment (Ehco1996) [inline dm/dm/master/openapi_view.go:81]: we should take care of the `https` because user may config tls connection between master

**Review by Ehco1996 — COMMENTED** [review]
rest LGTM

File: dm/dm/master/openapi_view_test.go [hunk 3]
```diff
@@ -334,9 +338,159 @@ func (s *OpenAPIViewSuite) TestRedirectRequestToLeader() {
 	s.Len(resultListSource.Data, 0)
 	s.Equal(0, resultListSource.Total)
 
-	// list source not from leader will get a redirect
-	result = testutil.NewRequest().Get(baseURL).GoWithHTTPHandler(s.T(), s2.openapiHandles)
-	s.Equal(http.StatusTemporaryRedirect, result.Code())
+	// list source not from leader will get result too
```
Comment (buchuitoudegou) [inline dm/dm/master/openapi_view_test.go:341]:
```suggestion
	// list source from non-leader will get result too
```

**Review by buchuitoudegou — APPROVED** [review]
rest LGTM

**Review by Ehco1996 — APPROVED** [review]
lgtm

**Pull request merged**